packages/wallet/src/services/AssetsTracker.ts

```diff
@@ -40,6 +42,7 @@ export const createAssetsTracker = (
     const sub = balanceTracker.utxo.total$
       .pipe(
         map(({ assets }) => [...(assets?.keys() || [])]),
+        distinctUntilChanged(deepEquals), // Optimize emitting if there aren't assets in total balance
```

Comment:
I think the comment should say it optimizes to not process duplicate emissions of the assets. It's not just no assets scenario

Reply:
The comment is updated with 629ec6b40a859b5223d2d92518814f6870225629